src/Raven.Server/Rachis/Follower.cs

```diff
@@ -483,7 +419,7 @@ private void NegotiateWithLeader(ClusterOperationContext context, LogLengthNegot
             {
                 KeepAliveAndExecuteAction(() =>
                 {
-                    onFullSnapshotInstalledTask = ReadAndCommitSnapshot(context, snapshot, cts.Token);
+                    onFullSnapshotInstalledTask = ReadAndCommitSnapshotAsync(context, snapshot, cts.Token).Result;
```

Comment:
maybe we don't need the `KeepAliveAndExecuteAction` now?

Reply:
We need it